src/QsCompiler/CompilationManager/AssemblyLoader.cs

```diff
@@ -204,9 +212,10 @@ private static (StringHandle, StringHandle)? GetAttributeType(MetadataReader met
         // have a single constructor taking a single string argument.
         private static (string, string)? GetAttribute(MetadataReader metadataReader, CustomAttribute attribute)
         {
-            var attrType = GetAttributeType(metadataReader, attribute);
-            QsCompilerError.Verify(attrType.HasValue, "the type of the custom attribute could not be determined");
-            var (ns, name) = attrType.Value;
+            var attrType =
+                GetAttributeType(metadataReader, attribute)
+                ?? throw new Exception("the type of the custom attribute could not be determined");
```

Comment:
Isn't this changing the current behavior?

Reply:
Sort of. It only changed what kind of exception is thrown. I changed it because the C# compiler did not realize that `QsCompilerError.Verify` would throw if the condition was false (so it thought `attrType` could still be null). However, I just learned that I can add `[<DoesNotReturnIf false>]` annotations to `Verify`, so I reverted this change.